IrynaFeshchenko/aws-shop
# Add configuration for task-2

What was done:
1. Link to deploys provided: 
 - Link to CloudFront - https://dfgkqtvwfk2uv.cloudfront.net/ 
 - Link to S3 - http://my-bucket-angular-feshchenko.s3-website.eu-central-1.amazonaws.com/
2. S3 bucket has been created and configured properly. The app has been uploaded to the bucket and is available though the Internet.
3. In addition to the previous work a CloudFront distribution is created and configured properly and the site is served now with CloudFront and is available through the Internet over CloudFront URL, not S3-website link (due to changes in bucket’s policy...). (Link to CloudFront website is provided. S3-website shows 403 Access Denied error.)
4. Serverless-finch and serverless-single-page-app-plugin plugins are added and configured. The app can be built and deployed by running npm script command.

In order to verify serverless solution:
 1. Checkout FE repo branch task-2
 2. Run npm run build command
 3. Configure inside serverless.yml file:
      - provider
      - custom.client.bucketname
      - service
 4. Deploy cloudformation stack via npm run cloudfront:deploy
 5. Deploy client assets via npm run client:deploy

To verify cloudfront invalidation:
 1. Run npm run build command
 2. Run npm run client:deploy command
 3. Run npm run cloudfront:invalidateCache command
